hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java

```diff
@@ -149,6 +160,16 @@
         LOG.info("Setting hoodie filter and realtime input format");
         combine.setHoodieFilter(true);
         combine.setRealTime(true);
+        if (job.get(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS, "").isEmpty()) {
+          List<String> partitions = new ArrayList<>(part.getPartSpec().keySet());
+          if (!partitions.isEmpty()) {
+            String partitionStr = String.join(",", partitions);
+            LOG.info("Setting Partitions in jobConf - Partition Keys for Path : " + path + " is :" + partitionStr);
+            job.set(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS, partitionStr);
+          } else {
+            job.set(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS, "");
```

Comment:
Do we need to call a set for else ?

Reply:
Yes, From what I understand, the input paths can be across tables (some partitioned and others not so). Hence, resetting before getSplits

Reply:
ok